core/src/consensus/light_client/mod.rs

```diff
@@ -0,0 +1,68 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pub mod verification;
+pub use self::verification::{verify_header, verify_state, verify_transaction};
+use ckey::SchnorrSignature;
+pub use ctypes::BlockNumber;
+pub use primitives::{H256, H512};
+use rlp::{Decodable, DecoderError, Encodable, Rlp, RlpStream};
+
+#[derive(PartialEq, Debug)]
+// (index_in_vset, sign); schnorr scheme
+pub struct Seal(Vec<(usize, SchnorrSignature)>);
+
+impl Encodable for Seal {
+    fn rlp_append(&self, s: &mut RlpStream) {
+        s.begin_list(self.0.len() * 2);
+        for (x, y) in self.0.iter() {
+            s.append(x);
+            s.append(&H512::from(*y));
+        }
+    }
+}
+
+impl Decodable for Seal {
+    fn decode(rlp: &Rlp<'_>) -> Result<Self, DecoderError> {
+        let item_count = rlp.item_count()?;
+        if item_count % 2 == 1 {
+            return Err(DecoderError::RlpInvalidLength {
+                expected: item_count + 1,
+                got: item_count,
+            })
+        }
+        let mut vec = Vec::with_capacity(item_count / 2);
+        for i in 0..(item_count / 2) {
+            let f: H512 = rlp.val_at(i * 2 + 1)?;
+            vec.push((rlp.val_at(i * 2)?, SchnorrSignature::from(f)));
+        }
+        Ok(Self(vec))
+    }
+}
+
+#[derive(RlpEncodable, RlpDecodable, PartialEq, Debug)]
+pub struct CoreHeader {
+    pub number: BlockNumber,
+    pub hash: H256,
+    pub next_validator_set_hash: H256,
+}
+
+#[allow(dead_code)]
+pub struct LightHeader {
+    pub core: CoreHeader,
+    pub state_root: H256,
+    pub transactions_root: H256,
+}
```

Comment:
If the `LightHeader` is not used, let's remove it.
And, let's change the name of the `CoreHeader` to `Header`.
Then we can understand the code easier.

@junha1 What do you think about it?